frontend/src/modules/member/components/bulk/bulk-edit-attribute-popover.vue

```diff
@@ -0,0 +1,382 @@
+  }
+};
+
+const fetchOrganizationsFn = (query, limit) => OrganizationService.listAutocomplete(query, limit)
+  .then((options) => options.map((o) => ({
+    ...o,
+    displayName: o.label,
+    name: o.label,
+    memberOrganizations: {
+      title: '',
+      dateStart: '',
+      dateEnd: '',
+    },
+  })))
+  .catch(() => []);
+
+const createOrganizationFn = (value) => OrganizationService.create({
+  name: value,
+})
+  .then((newOrganization) => ({
+    id: newOrganization.id,
+    label: newOrganization.displayName || newOrganization.name,
+    displayName: newOrganization.displayName || newOrganization.name,
+    name: newOrganization.displayName || newOrganization.name,
+    memberOrganizations: {
+      title: '',
+      dateStart: '',
+      dateEnd: '',
+    },
+  }))
+  .catch(() => null);
+
+const fetchCustomAttribute = (id) => MemberService.getCustomAttribute(id)
+  .then((response) => response.options.sort().map((o) => ({
+    id: o,
+    label: o,
+  })))
+  .catch(() => []);
+
+const updateCustomAttribute = (attribute, value) => {
+  const options = [...attribute.options];
+
+  options.push(value);
+
+  return MemberService.updateCustomAttribute(attribute.id, {
+    options,
+  }).then(() => ({
+    id: value,
+    label: value,
+  }));
+};
+
+const handleSubmit = async () => {
+  const formattedAttributes = getParsedAttributes(
+    computedCustomAttributes.value,
+    formModel.value,
+  );
+
+  // Remove any existent empty data
+  const data = {
+    ...formModel.value.joinedAt && {
+      joinedAt: formModel.value.joinedAt,
+    },
+    ...formModel.value.organizations.length && {
+      organizations: formModel.value.organizations.map(
+        (o) => ({
+          id: o.id,
+          name: o.name,
+          ...o.memberOrganizations?.title && {
+            title: o.memberOrganizations?.title,
+          },
+          ...o.memberOrganizations?.dateStart && {
+            startDate: o.memberOrganizations?.dateStart,
+          },
+          ...o.memberOrganizations?.dateEnd && {
+            endDate: o.memberOrganizations?.dateEnd,
+          },
+        }),
+      ).filter(
+        (o) => !!o.id,
+      ),
+    },
+    ...(Object.keys(formattedAttributes).length
+      || formModel.value.attributes) && {
+      attributes: {
+        ...(Object.keys(formattedAttributes).length
+          && formattedAttributes)
+      },
+    },
+  };
+
+  await store.dispatch('member/doBulkUpdateMembersAttribute', {
+    members: [...selectedMembers.value],
+    attributesToSave: data,
+  });
+
+  computedVisible.value = false;
+  emits('reload', true);
+
+  return null;
+};
+
+const handleDropdownClear = () => {
+  selectedAttribute.value = {};
+};
+
+const handleCancel = () => {
+  // clear form
+  formModel.value = {};
+  computedVisible.value = false;
+};
+
+</script>
+
+<script>
+export default {
+  name: 'AppBulkEditAttributePopover',
+};
+</script>
+
+<style>
+.custom-date-picker {
+  width: 100% !important;
+}
```

Comment:
The ".custom-date-picker" class is also being used in other components, so since this <style> is not scoped this styling may affect other components. Maybe add another class to the date picker of this component and here call it like .custom-date-picker._new-class_

But I don't think this class is working, at least it seems that the date picker is not occupying the input full width

Reply:
Thanks for pointing it out! I have updated the class name as you suggested so that it doesn’t affect other places. Also, the class was intended for the date-picker's input field and not the actual date-picker.